packages/cli/src/cliEntry.js

```diff
@@ -162,6 +164,27 @@ async function setupAndRun() {
 
   setProjectDir(ctx.root);
 
+  try {
+    // New version check must occur before `commander.parse` to ensure that
+    // the message of a new release happens before anything else.
```

Comment:
We don't need such guarantee. This check is low-priority and shouldn't block anything. Please print the message at the end of `setupAndRun` function (you can trigger the check in parallel earlier of course, just don't block it with `await`)

Reply:
The problem is that without blocking we cannot guarantee __when__ the message is printed.

Depending on the complexity of the command and time to fetch data from GH, the new version message might be printed before, during or at the end of a command. This might pollute the command's own logging flow and the user might also lose track of the new release message since it could appear whenever.

Reply:
so call it after the `await setupAndRun();`, it's all async and you can be sure all the output is already there in the console then

Reply:
I was wrong about it, sent a commit with a fix: cd5fa9f